docs/searching-algorithms/interpolation-search.mdx

```diff
@@ -0,0 +1,276 @@
+
+arr[] -> Array where elements need to be searched
+x     -> Element to be searched
+lo    -> Starting index in arr[]
+hi    -> Ending index in arr[]
+```
+
+## Algorithm
+
+- Calculate the value of `pos` using the probe position formula.
+- If it is a match, return the index of the item, and exit.
+- If the item is less than arr[pos], calculate the probe position of the left sub-array. Otherwise, calculate the same in the right sub-array.
+- Repeat Step 1 to Step 3 until a match is found or the size of the sub-array reduces to zero.
+
+## Code
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+<Tabs
+  defaultValue="py"
+  values={[
+    { label: 'Python', value: 'py', },
+    { label: 'C++', value: 'cpp', },
+  ]
+}>
+
+<TabItem value="py">
+
+```python
+# INTERPOLATION SEARCH
+
+# If the searching element is present in array,
+# then it returns index of it, else returns -1
+
+
+def interpolationSearch(arr, size, search_element):
+    # Assigning indexes
+    low = 0
+    high = size - 1
+    # As array is going to be sorted
+    # an element present in array must be in range defined by corner
+    while low <= high and search_element >=
+    arr[low] and search_element <= arr[high]:
+        if low == high:
+            if arr[low] == search_element:
+                return low
+            return -1
+
+        position = low + int(float(high - low) /
+        (arr[high] - arr[low])*(search_element - arr[low]))
+
+        # Condition of element found
+
+        if arr[position] == search_element:
+            return position
+
+        # If x is larger, x is in upper part
+
+        if arr[position] < search_element:
+            low = position + 1
+        else:
+            # If x is smaller, x is in lower part
+            high = position - 1
+
+    return -1
+
+
+# Starting with the main code
+# Creating array of the size n
+# Size would be taken into the variable n from user
+def main():
+    arr = []
+    size = int(input('Enter the size of array : '))
+
+    # Entering the elements in the array
+
+    print ('Enter elements :')
+    for index in range(0, size):
+        ele = int(input())
+        arr.append(ele)
+    print ('The elements entered are:\n', arr, '\n')
+
+    # Sorting our array
+
+    print ('The sorted array is: ')
+    arr.sort()
+    print (arr, '\n')
+
+    # Entering the searching element
+
+    search_element = int(input('Enter the element to be searched: '))
+
+    index = interpolationSearch(arr, size, search_element)
+
+    # If element was found
+
+    if index != -1:
+        # Returning the index of the element
+        print ('Element found at index', index)
+    else:
+        # Returning -1 as element has not found
+        print ('Element not found')
+main()
+```
+
+</TabItem>
+<TabItem value="cpp">
+
+```cpp
+//INTERPOLATION SEARCH
+#include<bits/stdc++.h> //Importing library (This is the special type of library which will include all basic libraries)
```

Comment:
Remove this `//Importing library (This is the special type of library which will include all basic libraries)`